### ISSUE TITLE
Incorrect exit code when errors encountered

### ISSUE THREAD

**Issue by greg-hellings** [ansible-community/ansible-lint#1081]
##### Summary
Specifying a config file on the command line appears to suppress exit codes when errors are found

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details
```console (paste below)
ansible --version
ansible 2.10.2
  config file = /home/ghelling/.ansible.cfg
  configured module search path = ['/home/ghelling/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/ghelling/venv/tools3/lib64/python3.7/site-packages/ansible
  executable location = /home/ghelling/venv/tools3/bin/ansible
  python version = 3.7.9 (default, Aug 19 2020, 17:05:11) [GCC 9.3.1 20200408 (Red Hat 9.3.1-2)]

ansible-lint --version
ansible-lint 4.3.5
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT
Fedora 31

##### STEPS TO REPRODUCE
* Run ansible-lint without config file option
* Observe exit code (2)
* Run same command with config file option
* Observe exit code (0)

```console
(tools3) [ghelling@work 🎩 kdump] (linting)$ cat ../../tests/ansible-lint 
skip_list:
  - "204"
(tools3) [ghelling@work 🎩 kdump] (linting)$ ansible-lint -R -r ../../tests/ansible_lint_rules .
[208] File permissions unset or incorrect
tasks/main.yml:9
Task/Handler: Edit kdump config file

You can skip specific rules or tags by adding them to your configuration file:                                        

┌────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│# .ansible-lint                                                                                                     │
│warn_list:  # or 'skip_list' to silence them completely                                                             │
│  - '208'  # File permissions unset or incorrect                                                                    │
│  - experimental  # all rules tagged as experimental                                                                │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
(tools3) [ghelling@work 🎩 kdump] (linting)$ echo $?
2
(tools3) [ghelling@work 🎩 kdump] (linting)$ ansible-lint -R -r ../../tests/ansible_lint_rules -c ../../tests/ansible-lint .
[208] File permissions unset or incorrect
tasks/main.yml:9
Task/Handler: Edit kdump config file

You can skip specific rules or tags by adding them to your configuration file:                                        

┌────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│# .ansible-lint                                                                                                     │
│warn_list:  # or 'skip_list' to silence them completely                                                             │
│  - experimental  # all rules tagged as experimental                                                                │
└────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
(tools3) [ghelling@work 🎩 kdump] (linting)$ echo $?
0
```

Running against [this PR](https://github.com/oasis-roles/ansible_collection_system/pull/35) in the roles/kdump folder

##### Desired Behaviour

When ansible-lint discovers an error, it should exit with non-zero

##### Actual Behaviour

Ansible lint exits with 0 when specifying a config file on the command line, but with non-zero when the config file is not specified

**Comment by duck-rh** [comment]
I'm experiencing the same problem and that's quite annoying for CI tests.

**Comment by netzvieh** [comment]
Okay so I tried to reproduce this but can't:

```yaml
---
- hosts: localhost
  roles:
    - name: blah
```

```
$ ansible-galaxy init /tmp/playbooks/roles/blah
$ ansible-lint -q /tmp/playbooks/playbook.yml 
[701] ../../../../../../../../tmp/playbooks/roles/blah/meta/main.yml:1
[703] ../../../../../../../../tmp/playbooks/roles/blah/meta/main.yml:1
[703] ../../../../../../../../tmp/playbooks/roles/blah/meta/main.yml:1
[703] ../../../../../../../../tmp/playbooks/roles/blah/meta/main.yml:1
$ echo $?
2
```

Creating `/tmp/playbooks/.ansible-lint`:

```yaml
---
skip_list:
  - '701'
```

```
$ ansible-lint -q -c /tmp/playbooks/.ansible-lint /tmp/playbooks/playbook.yml 
[703] ../../../../../../../../tmp/playbooks/roles/blah/meta/main.yml:1
[703] ../../../../../../../../tmp/playbooks/roles/blah/meta/main.yml:1
[703] ../../../../../../../../tmp/playbooks/roles/blah/meta/main.yml:1
$ echo $?
2
```

Any pointers on what I might be missing?

**Comment by netzvieh** [comment]
Interestingly I can reproduce #1102, even with the task inside a role, but I can't reproduce it with the 701/703 rule

ansible-lint shows 208, and skip_list contains 403 -> exit 0 which is wrong

ansible-lint shows 701, and skip_list contains 403 -> exit 2 which is correct

I'll see if I can find the culprit

**Comment by netzvieh** [comment]
This only applies to rules in the experimental category for now. I'd close this as a duplicate of #1102, as the reproducer there is simpler.
